qcodes/dataset/plotting.py

```diff
@@ -367,7 +367,7 @@ def _complex_to_real_preparser(alldata: NamedData,
         new_group = []
         new_groups: NamedData = [[], []]
         for index, parameter in enumerate(group):
-            data: np.ndarray = parameter['data']
+            data: np.ndarray = cast(np.ndarray, parameter['data'])
```

Comment:
can't the type be determined thanks to the cast?
```suggestion
            data = cast(np.ndarray, parameter['data'])
```